stdlib/tests/collections/smt.rs

```diff
@@ -0,0 +1,114 @@
+use crate::build_test;
+use test_utils::{
+    EmptySubtreeRoots, Felt, IntoBytes, MerkleStore, NodeIndex, Rpo256, StarkField, Word,
+    WORD_SIZE, ZERO,
+};
+
+#[test]
+fn smtget_opens_correctly() {
+    let empty = EmptySubtreeRoots::empty_hashes(64);
+    let initial_root = Word::from(empty[0]);
+    let seed = blake3::hash(b"some-seed");
+
+    // compute pseudo-random key/value pair
+    let key = <[u8; 8]>::try_from(&seed.as_bytes()[..8]).unwrap();
+    let key = u64::from_le_bytes(key);
+    let key = Felt::new(key);
+    let key = [key, key + Felt::new(1), key + Felt::new(2), key + Felt::new(3)];
+    let value = <[u8; 8]>::try_from(&seed.as_bytes()[8..16]).unwrap();
+    let value = u64::from_le_bytes(value);
+    let value = Felt::new(value);
+    let value = [value, value + Felt::new(1), value + Felt::new(2), value + Felt::new(3)];
```

Comment:
We can probably use `rand_value()`, `rand_array()` etc. from `rand-utils` crate to simplify this.

Reply:
One remark: I always try to avoid non-deterministic behavior on tests. The reason for that is the infamous "it works on my machine" problem as `rand::thread_rng` might randomly pull an input that will trigger a bug. Having that possibility discarded always helps when searching for a problem.

With that said, I think it's always good to pick one of the two: either static, hardcoded tests, or fuzzy. Given our library of choice (proptest), if we want to have random inputs, I think it's better to use this framework. It will record failed tests and their inputs, if we come to that - so we never have the "it works on my machine" problem.

I updated both tests to take inputs from `proptest`. As future improvement, we can consider implementing [Arbitrary](https://altsysrq.github.io/rustdoc/proptest/1.0.0/proptest/arbitrary/trait.Arbitrary.html) for `Word` and `Felt` so we just pass them to proptest arguments instead of seeding/generating

Reply:
I'm not sure I would have gone with a `proptest` in this case - using it could be pretty heavy and in this case, I'm not sure we gain much from testing with different seeds. But since it is already there - let's keep it.

> I updated both tests to take inputs from `proptest`. As future improvement, we can consider implementing [Arbitrary](https://altsysrq.github.io/rustdoc/proptest/1.0.0/proptest/arbitrary/trait.Arbitrary.html) for `Word` and `Felt` so we just pass them to proptest arguments instead of seeding/generating

For `Word` we could do this probably quite easily, but for `Felt` this would need to be done in Winterfell, right?